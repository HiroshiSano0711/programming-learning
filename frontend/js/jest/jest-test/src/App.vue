<template>
  <div id="app">
    <MessageList :messages="messages" />
  </div>
</template>

<script>
import MessageList from './components/MessageList'

export default {
  name: 'App',
  components: {
    MessageList
  },
  data: () => {
    return {
      messages: ['Hey John', 'Howdy Paco']
    }
  }
}
</script>
