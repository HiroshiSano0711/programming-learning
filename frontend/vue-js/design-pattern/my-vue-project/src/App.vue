<template>
  <div id="app">
    <button @click="showAlert">アラートが出るよ</button>
  </div>
</template>

<script>
export default {
  name: 'app',
  data() {
    return {
      message: '警告メッセージ'
    }
  },
  methods: {
    showAlert() {
      alert(this.message)
    }
  },
}
</script>

<style>

</style>
