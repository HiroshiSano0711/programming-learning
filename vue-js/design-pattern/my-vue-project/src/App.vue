<template>
  <div id="app">
    <button @click="showAlert">アラートが出るよ</button>
  </div>
</template>

<script>
export default {
  name: 'app',
  data () {
    return {
      message: 'Hello, World'
    }
  },
  methods: {
    showAlert() {
      alert(this.message)
    }
  },
}
</script>

<style>

</style>
