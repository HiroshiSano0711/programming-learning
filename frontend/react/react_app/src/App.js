import React, { Component } from 'react'
import './App.css';

class App extends Component {
  constructor(props){
    super(props)
    this.state = {
      msg: 'count start',
      counter: 0,
      flg: true,
    }
    this.doAction = this.doAction.bind(this)
  }

  doAction(e) {
    this.setState({
      msg: this.state.counter,
      counter: this.state.counter + 1,
      flg: !this.state.flg,
    })
  }

  render() {
    return <div className="App">
      <header className="App-header">
        <p>{this.state.msg}</p>
        <p>{this.state.count}</p>
        <button onClick={this.doAction}>click</button>
      </header>
    </div>
  }
}

export default App;
