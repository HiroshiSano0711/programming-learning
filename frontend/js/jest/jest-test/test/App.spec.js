// import { mount } from '@vue/test-utils';
// import MessageList from '../src/components/MessageList';

// describe('MessageList.test.js', () => {
//   let cmp;

//   beforeEach(() => {
//     cmp = mount(MessageList, {
//       propsData: { messages: ['Cat'] }
//     });
//   });

//   it('has received Cat as the message property', () => {
//     expect(cmp.vm.messages).toEqual(['Cat'])
//   });

//   it('has the expected html structure', () => {
//     expect(cmp.element).toMatchSnapshot()
//   });
// });

import { shallowMount } from "@vue/test-utils";
import App from "../src/App";

describe("App.test.js", () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(App, { // Create a shallow instance of the component
      data: ()=> {
        return {
          messages: ["Cat"]
        }
      }
    }); 
  });

  it('messagesは["Cat"]と等しいこと', () => { // Within cmp.vm, we can access all Vue instance methods
    expect(cmp.vm.messages).toEqual(["Cat"]);
  });

  it("期待されるHTML構造を持っていること", () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
