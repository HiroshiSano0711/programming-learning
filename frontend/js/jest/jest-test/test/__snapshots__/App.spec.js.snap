// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <messagelist-stub
    messages="Cat"
  />
</div>
`;

exports[`App.test.js 期待されるHTML構造を持っていること 1`] = `
<div
  id="app"
>
  <messagelist-stub
    messages="Cat"
  />
</div>
`;
